src/tlp_lib/smartcontracts/MockSC.py

```diff
@@ -0,0 +1,50 @@
+from datetime import datetime
+from typing import Self
+
+from tlp_lib.protocols import GMITLP_Encrypted_Message, TLP_Digest, TLP_Digests
+from tlp_lib.smartcontracts.protocols import SC_Coins, SC_ExtraTime, SC_Solutions, SC_UpperBounds
+
+
+class MockSC:
+    commitments: TLP_Digests
+    start_time: int
+    upper_bounds: SC_UpperBounds
+    coins: SC_Coins
+    solutions: SC_Solutions
+    initial_timestamp: int
+    helper_id: int
+    extra_time: SC_ExtraTime
+
+    def initiate(
```

Comment:
why use a `initiate` method instead of passing the class and instantiating it when needed (instead of calling `initiate`)?

Reply:
The idea was to imitate Smart Contracts outside of DFMITLP due to the instantiation being very different from contract to contract (and requiring a different set of arguments, which might be too convoluted to pass into DFMITLP). 

The Mock contract does not have this issue, however, I had to make something universal for all contracts as they should implement the same interface. And `initiate` was this universal function I required all contracts to implement, while each has the freedom to define their own __init__ with their own set of arguments to prepare for instantiation.

Reply:
OK, that makes sense. So it's like a factory, cool.

Reply:
Exactly, technically, the SC is a factory